src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/EndpointFactory.cs

```diff
@@ -0,0 +1,86 @@
+using NuGet.Services.Metadata.Catalog.Monitoring.Utility;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test
+{
+    /// <summary>
+    /// Creates <see cref="Endpoint"/>s by name. 
+    /// </summary>
+    public class EndpointFactory
+    {
+        public EndpointFactory(
+            ValidatorFactory validationFactory, 
+            Func<string, Uri> endpointCursorUriFactory, 
+            Func<HttpMessageHandler> messageHandlerFactory, 
+            ILoggerFactory loggerFactory)
+        {
+            _validationFactory = validationFactory;
+            _endpointCursorUriFactory = endpointCursorUriFactory;
+            _messageHandlerFactory = messageHandlerFactory;
+            _loggerFactory = loggerFactory;
+        }
+
+        /// <summary>
+        /// Creates the <see cref="Endpoint"/> for which <see cref="Type.Name"/> is equal to <paramref name="endpointName"/>.
+        /// </summary>
+        public Endpoint Create(string endpointName)
+        {
+            try
+            {
+                /// Find the <see cref="Endpoint"/> named after <param name="endpointName"/>
+                var endpointType = Assembly
+                    .GetExecutingAssembly()
+                    .GetTypes()
+                    .FirstOrDefault(t => t.Name == endpointName);
+
+                /// Create a <see cref="HttpReadCursor"/> from the <see cref="Uri"/> for this <see cref="Endpoint"/>.
+                var endpointCursorUri = _endpointCursorUriFactory(endpointName);
+                var endpointCursor = new HttpReadCursor(endpointCursorUri, _messageHandlerFactory);
+
+                /// Construct the <see cref="Endpoint"/>.
+                var loggerType = typeof(ILogger<>).MakeGenericType(endpointType);
+
+                var endpointConstructor = endpointType
+                    .GetConstructor(new Type[] 
+                    {
+                        typeof(ReadCursor),
+                        typeof(ValidatorFactory),
+                        loggerType
+                    });
+
+                if (endpointConstructor == null)
+                {
+                    throw new ArgumentException($"Endpoint must have a constructor with a " +
+                        $"{nameof(ReadCursor)}, " +
+                        $"{nameof(ValidatorFactory)}, and " +
+                        $"{loggerType.Name}!", nameof(endpointName));
+                }
+
+                var endpoint = endpointConstructor.Invoke(new object[] 
+                {
+                    endpointCursor,
+                    _validationFactory,
+                    _loggerFactory.CreateTypedLogger(endpointType)
+                });
+
+                return endpoint as Endpoint;
+            }
+            catch (Exception e)
+            {
+                throw new Exception($"Failed to create endpoint named {endpointName}!", e);
+            }
+        }
+
+        private ValidatorFactory _validationFactory;
```

Comment:
in all files, please move properties and fields to the start of the class, according to our coding conventions